File: gadapter/src/main/java/com/zeyad/gadapter/fastscroll/viewprovider/DefaultBubbleBehavior.java
package com.zeyad.gadapter.fastscroll.viewprovider;

class DefaultBubbleBehavior implements ViewBehavior {

    private final VisibilityAnimationManager animationManager;

    DefaultBubbleBehavior(VisibilityAnimationManager animationManager) {
        this.animationManager = animationManager;
    }

    @Override
    public void onHandleGrabbed() {
        animationManager.show();
    }

    @Override
    public void onHandleReleased() {
        animationManager.hide();
    }

    @Override
    public void onScrollStarted() {
    }

    @Override
    public void onScrollFinished() {
    }
}

File: gadapter/src/main/java/com/zeyad/gadapter/fastscroll/viewprovider/DefaultBubbleBehavior.kt
package com.zeyad.gadapter.fastscroll.viewprovider

internal class DefaultBubbleBehavior(private val animationManager: VisibilityAnimationManager) : ViewBehavior {

    override fun onHandleGrabbed() {
        animationManager.show()
    }

    override fun onHandleReleased() {
        animationManager.hide()
    }

    override fun onScrollStarted() {}

    override fun onScrollFinished() {}
}
